testsuite/tests/typing-modules/inclusion_errors.ml

```diff
@@ -1051,6 +1156,76 @@ Error: Signature mismatch:
          type t = [ `A of float ]
        is not included in
          type t = private [> `A of int ]
+       The type float is not equal to the type int
+|}];;
+
+module M : sig
+  type t = private [< `A | `B]
+end = struct
+  type t = private [`A | `B]
+end;;
+[%%expect{|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type t = private [`A | `B]
+5 | end..
+Error: Signature mismatch:
+       Modules do not match:
+         sig type t = private [ `A | `B ] end
+       is not included in
+         sig type t = private [< `A | `B ] end
+       Type declarations do not match:
+         type t = private [ `A | `B ]
+       is not included in
+         type t = private [< `A | `B ]
+       The type [ `A | `B ] is not equal to the type [< `A | `B ]
+       The tag `B is guaranteed to be present in the first variant type,
+       but not in the second
+|}];;
+
+module M : sig
+  type t = [`A | `B]
+end = struct
+  type t = private [`A | `B]
+end;;
+[%%expect{|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type t = private [`A | `B]
+5 | end..
+Error: Signature mismatch:
+       Modules do not match:
+         sig type t = private [ `A | `B ] end
+       is not included in
+         sig type t = [ `A | `B ] end
+       Type declarations do not match:
+         type t = private [ `A | `B ]
+       is not included in
+         type t = [ `A | `B ]
+       A private type would be revealed.
```

Comment:
This sounds right.  What would you suggest for the case in `typing-modules/extension_constructors_errors_test.ml` where we have

```ocaml
module M1 : sig type t += A end = struct type t += private A end;;
[%%expect{|
Line 1, characters 34-64:
1 | module M1 : sig type t += A end = struct type t += private A end;;
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Signature mismatch:
       Modules do not match:
         sig type t += private A end
       is not included in
         sig type t += A end
       Extension declarations do not match:
         type t += private A
       is not included in
         type t += A
       A private type would be revealed.
|}];;
```

?  I was considering "A private extension constructor would be revealed", but I'm not sure of the correct language.

Reply:
Since it is the definition that is private (`type t += private A | B` but not `type t += A | private B`), I would propose to use the same message that for (non-extensible) private variant definitions.